### DISCUSSION TITLE
illegal unregister of arrival

### DISCUSSION THREAD

**Discussion by DanielaRichter** [r-simmer/simmer#266] [Q&A]
Hi,
I have my resources on a schedule. To prevent tasks from starting when they cannot be finished on the same workday, I added a dummy resource that ends its schedule the relevant time before the resource itself. by requiring both resources to be seized, the dummy should determine the schedule of the activity.

The simulation finishes for the first batch of arrivals. The second batch of arrivals however get the error message: illegal unregister of arrival.

I assume the problem is that the dummy is seized and waiting for the other resource (which is currently busy), then the capacity of the dummy drops. When the other resource becomes available, the dummy is not available anymore. However, I was not able to recreate the error in a simpler model. When I use a resource and a blocker resource, the simulation just finishes without an error.

Could someone explain the meaning of the error message?
Alternatively, is there another way to prevent a "seize" when the following timeout cannot be finished until the end of the schedule?

**Comment by Enchufa2** [comment]
Not sure if I understand the setup. There must be a bug somewhere, because that error shouldn't be reached. Are there clones involved here?

**Reply by DanielaRichter** [reply]
Yes there are ;)

below I post the main trajectory that assembles individual trajectories, which include branches. The resources in those trajectories are on a schedule and multiple resources are sometimes seized in one activity. But no further clones in those sub-trajectories.
In the last run, 11 patients finished the trajectory before the error occured for batch2 in the RNA_libprep sub-trajectory.

```
### main trajectory patient ### 
patient <- trajectory() %>% 
  set_attribute("start", now(envs)) %>%
  #log_("arrival") %>% 
  clone (n = 2,
         tumor_acquisition,
         germline_aquisition) %>% 
  synchronize(wait = TRUE) %>% 
  #log_("tumor and germline sample arrived") %>% 
  batch(BATCHSIZE) %>% 
  clone (n = 3,
         germline_extraction,
         join(tumor_dissection_DNA, tumorDNA_extraction),
         join(tumor_dissection, tumorRNA_extraction)) %>% 
         synchronize(wait = TRUE) %>% 
  #log_("Extraction ready") %>% 
  clone( n = 2,
         DNA_libprep,
         RNA_libprep) %>% 
  synchronize(wait = TRUE) %>% 
  #log_("Libprep ready") %>% 
  join(sequencing) %>% 
  join(bioinformatics) %>%
  simmer::separate() %>% 
  join(upload) %>%
  #log_("upload finished")
  set_attribute("end", now(envs))
```

**Reply by DanielaRichter** [reply]
okay, I found it!
I commented out all resources but one per activity. This works fine. No error message.
now I am putting the resources back one by one. Until the RNA_libprep, no problem. Then I found a typo in the RNA_libprep causing a double seize.

I am sorry, but the error message had me on the wrong path. SOLVED! Problem in front of the keyboard ;)

**Reply by Enchufa2** [reply]
Great! Nothing to be sorry about. :)
